diff -Naur d3des/d3des.c ../../../anet-1.0/src/3rdparty/d3des/d3des.c
--- d3des/d3des.c	Mon Nov 30 08:13:14 1992
+++ ../../../anet-1.0/src/3rdparty/d3des/d3des.c	Wed Jun 20 11:39:25 2001
@@ -14,7 +14,7 @@
 
 #include "d3des.h"
 
-static void scrunch(unsigned char *, unsigned long *);
+static void scrunch(const unsigned char *, unsigned long *);
 static void unscrun(unsigned long *, unsigned char *);
 static void desfunc(unsigned long *, unsigned long *);
 static void cookey(unsigned long *);
@@ -56,7 +56,7 @@
 	43, 48, 38, 55, 33, 52, 45, 41, 49, 35, 28, 31 };
 
 void deskey(key, edf)	/* Thanks to James Gillogly & Phil Karn! */
-unsigned char *key;
+const unsigned char *key;
 short edf;
 {
 	register int i, j, l, m, n;
@@ -147,7 +147,7 @@
 	}
 
 static void scrunch(outof, into)
-register unsigned char *outof;
+register const unsigned char *outof;
 register unsigned long *into;
 {
 	*into	 = (*outof++ & 0xffL) << 24;
@@ -398,7 +398,7 @@
 #ifdef D2_DES
 
 void des2key(hexkey, mode)		/* stomps on Kn3 too */
-unsigned char *hexkey;			/* unsigned char[16] */
+const unsigned char *hexkey;			/* unsigned char[16] */
 short mode;
 {
 	short revmod;
@@ -412,7 +412,8 @@
 	}
 
 void Ddes(from, into)
-unsigned char *from, *into;		/* unsigned char[8] */
+const unsigned char *from;		/* unsigned char[8] */
+unsigned char *into;		/* unsigned char[8] */
 {
 	unsigned long work[2];
 
@@ -425,7 +426,7 @@
 	}
 
 void D2des(from, into)
-unsigned char *from;			/* unsigned char[16] */
+const unsigned char *from;			/* unsigned char[16] */
 unsigned char *into;			/* unsigned char[16] */
 {
 	unsigned long *right, *l1, swap;
diff -Naur d3des/d3des.h ../../../anet-1.0/src/3rdparty/d3des/d3des.h
--- d3des/d3des.h	Mon Nov 30 08:13:14 1992
+++ ../../../anet-1.0/src/3rdparty/d3des/d3des.h	Wed Jun 20 11:36:54 2001
@@ -27,7 +27,7 @@
 	unsigned char byte[8];
 	} M68K;
 
-extern void deskey(unsigned char *, short);
+extern void deskey(const unsigned char *, short);
 /*		      hexkey[8]     MODE
  * Sets the internal key register according to the hexadecimal
  * key contained in the 8 bytes of hexkey, according to the DES,
@@ -55,7 +55,7 @@
 #ifdef D2_DES
 
 #define desDkey(a,b)	des2key((a),(b))
-extern void des2key(unsigned char *, short);
+extern void des2key(const unsigned char *, short);
 /*		      hexkey[16]     MODE
  * Sets the internal key registerS according to the hexadecimal
  * keyS contained in the 16 bytes of hexkey, according to the DES,
@@ -63,14 +63,14 @@
  * NOTE: this clobbers all three key registers!
  */
 
-extern void Ddes(unsigned char *, unsigned char *);
+extern void Ddes(const unsigned char *, unsigned char *);
 /*		    from[8]	      to[8]
  * Encrypts/Decrypts (according to the keyS currently loaded in the
  * internal key registerS) one block of eight bytes at address 'from'
  * into the block at address 'to'.  They can be the same.
  */
 
-extern void D2des(unsigned char *, unsigned char *);
+extern void D2des(const unsigned char *, unsigned char *);
 /*		    from[16]	      to[16]
  * Encrypts/Decrypts (according to the keyS currently loaded in the
  * internal key registerS) one block of SIXTEEN bytes at address 'from'
